### PR TITLE
fix tests

### LINKED ISSUE #1172: examples: investigate "fuzzy" image comparisons
For our Paraview and obspy tests we generate images and compare them against a known good image (I believe the term is a gold file) using `cmp`. This appears to be brittle when generating images across machines. We should investigate a fuzzy matching solution so we tolerate a small amount of difference.

### LINKED ISSUE #1266: `ch-test`: use locale `en_US.utf8` if available
Locale `C` causes problems on Python 3.6; see issues #970 and #1262, which in turn causes the test suite to fail.

### LINKED ISSUE #1288: examples: images using centos:8 are failing
Since centos 8 is EOL we need to migrate this images.

related #1279

### LINKED ISSUE #1289: examples: tests that revert images after "cray-ifying" them still use `ch-tar2dir`
`ch-tar2dir` is deprecated and should be replaced by `ch-convert`. This leads to failures in the tests because the initial unpack is done with `ch-convert` which doesn't place the sentinel file that `ch-tar2dir` expects.

### LINKED ISSUE #1290: examples: serial MPI tests fail on Cray xc40
After some recent changes serial MPI tests fail on our Cray xc40. It sounds like this is the way that things will be moving forward, we should skip these tests.

### PR DESCRIPTION
**Pull request by reidpr**
This works around locale issues for the test suite (#1262 and possibly #970, closes #1266) and adds fuzzy image compare (closes #1172).
Additionally, this PR closes #1290, closes #1289, and closes #1288.

### CODE REVIEW

File: test/common.bash [hunk 1]
```diff
@@ -116,6 +116,34 @@ pedantic_fail () {
...
+    sample=$2
+    pixel_max_ct=${3:-0}
+    sample_base=${sample%.*}
+    sample_ext=${sample##*.}
+    diff_=${sample_base}.diff.${sample_ext}
+    echo "reference:   ${ref}"
+    echo "sample:      ${sample}"
+    echo "diff image:  ${diff_}"
+    # See: https://imagemagick.org/script/command-line-options.php#metric
+    pixel_ct=$(compare -metric AE "$ref" "$sample" "$diff_" 2>&1 || true)
```
Comment (heasterday) [inline test/common.bash:135]: Should ImageMagick be add as a dependency for the "complete test suite" in `configure.ac`?

**Review by heasterday — COMMENTED** [review]
I would like to test this on my dev machine but this looks good. One minor comment.

File: test/common.bash (hunk 1 above)
Comment (reidpr) [reply test/common.bash:135]: Yeah probably. There's a lot of that stuff. Why don't we defer it to issue #1267.

**Comment by heasterday** [comment]
@reidpr It's not allowing me to request your review, can you take a look at this PR?

**Comment by heasterday** [comment]
@reidpr Updated issues that this resolves. @j-ogas I would appreciate you taking a look at this one.

**Review by j-ogas — APPROVED** [review]

**Pull request merged**